### ISSUE TITLE
Allow duplicate folder name creation and use

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#50]
**Original reporter**: *flabrie@users.sourceforge.net*

Since YAM 2.4 now support hierachical folders, it would be nice to allow duplicate folder name. Since I could now classify my folder in category, I would like to organize them logically, like:

```
+ Mac OS X
| + News
| + Maling Lists
| + ...
+ AmigaOS
+ News
+ Mailing lists
+ ...
```
Right now it's impossible. It would be necessary to generate an hidden unique ID or use the full path reference to associate filter move function to folder instead of just the folder name.

By the way, this reference to an hidden ID would be far better and would be more flexible than name reference. When a user will change a folder name, it would not be necessary to change all filter associated to this folder.

**Comment by jens-maus** [comment]
Makes sense and is in line with what wew need for full IMAP usage later on anyway. We should really allow to have folder with equal names, thus differentiate between display name and folder path.

**Comment by jens-maus** [comment]
**Original reporter**: *@tboeckel*

(In [7307]) * YAM_FO.c, Config.c, FolderList.c, misc: each folder now gets a unique ID assigned. In the future everything will be changed to reference folders by this ID only instead of referencing them by name, path or type. This will eventually make it possible to have equally named folders in different paths. This refs #400 and also refs #50.

**Comment by jens-maus** [comment]
Now that we have the functionality in YAM to refer to a folder based on a ID we should definitely  implement this for 2.10

**Comment by jens-maus** [comment]
**Original reporter**: *@tboeckel*

In [changeset:"8171"]:
```
#CommitTicketReference repository="" revision="8171"
* mui/FolderEditWindow.c: removed the restriction of unique folder names. This makes it possible to have equally named folders. This closes #50.
```
